### PR TITLE
Fix Filter reload Issues with Live TV

### PR DESCRIPTION
**Pull request by candry7731**
Add the ability for the user to say "all" to show all channels after the guide has been filtered by a voice command.

In the channel view the reload button will clear the filter and show all channels

### CODE REVIEW

File: components/liveTv/schedule.brs [hunk 1]
```diff
@@ -279,6 +286,17 @@ end sub
 
 function onKeyEvent(key as string, press as boolean) as boolean
     if not press then return false
+    print "Key Pressed = "key
```
Comment (1hitsong) [inline components/liveTv/schedule.brs]: Debug Code

File: components/liveTv/schedule.brs [hunk 2]
```diff
@@ -279,6 +286,17 @@ end sub
...
+
+    if key = "replay"
+        m.spinner.visible = true
+        m.top.searchTerm = " "
+        m.LoadChannelsTask.searchTerm = " "
+        channelsearchTermSet()
+        print "Press Return"
```
Comment (1hitsong) [inline components/liveTv/schedule.brs]: Debug code

**Review by 1hitsong — CHANGES_REQUESTED** [review]
Voice search is activating Roku search for me, instead of the in-app JF search.

**Comment by 1hitsong** [comment]
👍 Works as described.

Can we apply the same logic to both views? Add reload button logic to the TV Guide, and "All" voice filter to the Channels view?

**Review by 1hitsong — APPROVED** [review]

**Comment by jimdogx** [comment]
A couple of things:

1. LoadItemTask2.xml (not the BRS), has a default value of "All" (uppercase).  Not sure if that matters but thought I should point it out since you are specifically looking for lowercase "all"

2. The replay button doesn't work for me.  It's not even registering on the TV Guide (I added some debug print statements and "replay" is not even showing up in onKeyEvent).  No idea why or if it's just me.

Voice filtering and resetting the filter by saying "All" does work though 👍

**Comment by candry7731** [comment]
> A couple of things:
> 
> 1. LoadItemTask2.xml (not the BRS), has a default value of "All" (uppercase).  Not sure if that matters but thought I should point it out since you are specifically looking for lowercase "all"
> 2. The replay button doesn't work for me.  It's not even registering on the TV Guide (I added some debug print statements and "replay" is not even showing up in onKeyEvent).  No idea why or if it's just me.

_the replay button is assigned by roku to take users back to the current time when in the guide view. so if you scroll to the right to advance the time in the guide and then hit the replay button it will bring you back to the current time. _
> 
> Voice filtering and resetting the filter by saying "All" does work though 👍

**Comment by jimdogx** [comment]
@candry7731 I'm talking about this piece of code:

```
   if key = "replay"
        m.spinner.visible = true
        m.top.searchTerm = " "
        m.LoadChannelsTask.searchTerm = " "
        channelsearchTermSet()
        return true
    end if
```

I thought it implied that I could say "All" or that I could hit the replay button to remove the filter.  But this code is never hit.  So maybe it's not needed?  Or am I not using it correctly?

File: components/liveTv/schedule.brs [hunk 3]
```diff
@@ -279,6 +286,15 @@ end sub
 
 function onKeyEvent(key as string, press as boolean) as boolean
     if not press then return false
+
+    if key = "replay"
```
Comment (jimdogx) [inline components/liveTv/schedule.brs]: Remove this if it doesn't do anything (see conversation).

**Review by jimdogx — CHANGES_REQUESTED** [review]

File: components/liveTv/schedule.brs (hunk 3 above)
Comment (candry7731) [reply components/liveTv/schedule.brs]: Done, This is removed

**Review by jimdogx — APPROVED** [review]

**Pull request merged**